products/compute/api.yaml

```diff
@@ -1285,6 +1285,67 @@ objects:
           An optional textual description of the valid disk size, such as
           "10GB-10TB".
         output: true
+  - !ruby/object:Api::Resource
+    name: 'DiskResourcePolicyAttachment'
+    input: true
+    base_url: projects/{{project}}/zones/{{zone}}/disks/{{disk}}
+    create_verb: :POST
+    create_url: projects/{{project}}/zones/{{zone}}/disks/{{disk}}/addResourcePolicies
+    delete_verb: :POST
+    delete_url: projects/{{project}}/zones/{{zone}}/disks/{{disk}}/removeResourcePolicies
+    self_link: projects/{{project}}/zones/{{zone}}/disks/{{disk}}
+    nested_query: !ruby/object:Api::Resource::NestedQuery
+      keys:
+        - resourcePolicies
+      is_list_of_ids: true
+    identity:
+      - name
+    description: |
+      Disk resource policies define a schedule for taking snapshots and a
+      retention period for these snapshots.
+    async: !ruby/object:Api::Async
+      operation: !ruby/object:Api::Async::Operation
+        kind: 'compute#operation'
+        path: 'name'
+        base_url: 'projects/{{project}}/zones/{{zone}}/operations/{{op_id}}'
+        wait_ms: 1000
+        timeouts: !ruby/object:Api::Timeouts
+          insert_minutes: 3
+      result: !ruby/object:Api::Async::Result
+        path: 'targetLink'
+      status: !ruby/object:Api::Async::Status
+        path: 'status'
+        complete: 'DONE'
+        allowed:
+          - 'PENDING'
+          - 'RUNNING'
+          - 'DONE'
+      error: !ruby/object:Api::Async::Error
+        path: 'error/errors'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'disk'
+        resource: 'Disk'
+        imports: 'name'
+        description: |
+          The name of the disk in which the resource policies are attached to.
+        required: true
+        url_param_only: true
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'zone'
```

Comment:
In `terraform.yaml` do you mind setting `zone` `required: false` and `default_from_api: true`? The provider-level zone will automatically get inferred that way.